### PR TITLE
CN-7310: created crypto util tests

### PR DESCRIPTION
**Pull request by srkela**
## Checklist (Mandatory):

- [x] I have named the PR correctly (Check PR1)
- [x] Check the commit’s or even all commits message styles matches our requested structure. (issue number, etc.) (Check CR10)
- [x] I have tested new code locally (Check CR3)
- [x] I have done a smoke test of the larger feature locally (Check CR3)
- [x] I have added at least two reviewers (Check PR5)
- [x] I have assigned the PR to myself (Check PR4)
- [x] I have merged develop (parent) branch into my feature branch and resolved all conflicts (Check PR7)

## Checklist (Optional):

- [ ] I have added necessary documentation, including instructions how to test (Check CR8)
- [ ] I have added tests that prove my fix/implementation is effective or that my feature works
- [ ] I have written the code according to the adopted style guide? (Check CR7)

## Additional notes

> What does this implement/fix? Explain your changes. (optional)
> Are there any scripts that need running (eg. yarn install)?
> Are there any edge cases that need checking?
> Are there any specific test instructions?

[MVPWorkshop Pull Request & Code Review Guidelines](https://docs.google.com/document/d/1TcuU1JBhlGZLtpy0osSNinXQ2piP_yrDuRcXdqaOS_U/edit#heading=h.5b6kpdr4btq8)

### CODE REVIEW

File: __tests__/crypto-util.test.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import cryptoUtil from "../app/utils/crypto-util";
+import cryptoUtil from "../app/utils/crypto-util/crypto-util";
```
Comment (fjovakaric) [inline __tests__/crypto-util.test.js:1]: should remove the file

File: app/utils/crypto-util/crypto-util.test.js [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+beforeEach(() => {
+  mockState = _.cloneDeep(initialState);
+});
+afterEach(() => {
+  mockState = _.cloneDeep(initialState);
+});
+
+describe("isErc20()", () => {
+  it("should return true coin is ERC20", () => {
```
Comment (fjovakaric) [inline app/utils/crypto-util/crypto-util.test.js:32]: should go through array with all supported erc20 coins

File: app/utils/crypto-util/crypto-util.test.js [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+    expect(cryptoUtil.isERC20("ETH")).toBeTruthy();
+  });
+
+  it("should return true coin is ERC20", () => {
+    expect(cryptoUtil.isERC20("BTC")).toBeFalsy();
```
Comment (fjovakaric) [inline app/utils/crypto-util/crypto-util.test.js:37]: should go through array of all non erc20 coins

File: app/utils/crypto-util/crypto-util.test.js [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+    expect(cryptoUtil.hasLinkToBuy("DAI")).toBeTruthy();
+  });
+  it("should return true if coin has not link to buy", () => {
+    expect(cryptoUtil.hasLinkToBuy("BTC")).toBeFalsy();
+  });
+});
+
+describe("buyInApp()", () => {
+  it("should return true if coin is available to buy", () => {
+    expect(cryptoUtil.buyInApp("BTC")).toBeTruthy();
```
Comment (fjovakaric) [inline app/utils/crypto-util/crypto-util.test.js:63]: should go through array for all coins in compliance

File: app/utils/crypto-util/crypto-util.test.js [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+  });
+  it("should return true if coin is available to buy", () => {
+    expect(cryptoUtil.buyInApp("DAI")).toBeFalsy();
```
Comment (fjovakaric) [inline app/utils/crypto-util/crypto-util.test.js:66]: should go through array of all unsupported coins in compliance

File: app/utils/crypto-util/crypto-util.test.js [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+  });
+});
+
+describe("provideLink()", () => {
+  it("should return should return string https://buy.moonpay.io/celsius", () => {
+    expect(cryptoUtil.provideLink("DAI")).toBe(
+      "https://buy.moonpay.io/celsius"
```
Comment (fjovakaric) [inline app/utils/crypto-util/crypto-util.test.js:73]: move all coin links to a constant

File: app/utils/crypto-util/crypto-util.test.js [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+      "https://www.dash.org/where-to-buy"
+    );
+  });
+  it("should return should return null if anything else", () => {
+    expect(cryptoUtil.provideLink("")).toBeNull();
+  });
+});
+
+describe("provideText()", () => {
+  const btc = "BTC";
```
Comment (fjovakaric) [inline app/utils/crypto-util/crypto-util.test.js:107]: use `let coin = "BTC"`, not a separate variable for each coin. this should be inside the test

File: app/utils/crypto-util/crypto-util.test.js [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+    expect(cryptoUtil.provideText(omg)).toBe(`Buy ${omg} on MoonPay`);
+  });
+
+  it("should return should return null if anything else", () => {
+    expect(cryptoUtil.provideText("")).toBeNull();
+  });
+});
+
+describe("getBlockExplorerLink()", () => {
+  const transactionBtc =
```
Comment (fjovakaric) [inline app/utils/crypto-util/crypto-util.test.js:148]: should be `transactionId = ...`. can add anything, no need for correct id from mocks

File: app/utils/crypto-util/crypto-util.test.js [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+    mockTransactionsStore.postman13.transactionList[
+      "677c1e9e-7ce2-4a7c-927b-089d778063cb"
+    ];
+  it("should return transaction link and blockexplorer name in string", () => {
```
Comment (fjovakaric) [inline app/utils/crypto-util/crypto-util.test.js:152]: all test names are the same, this Is confusing

**Review by fjovakaric — CHANGES_REQUESTED** [review]

File: app/utils/crypto-util/crypto-util.test.js [hunk 10]
```diff
@@ -0,0 +1,227 @@
...
+    coinForText = null;
+  });
+  it(`should return string Buy coin in App"`, () => {
+    coinForText = ["BTC", "BCH", "ETH", "LTC", "XRP", "CEL", "XLM"];
+    coinForText.forEach(c => {
+      expect(cryptoUtil.provideText(c)).toBe(`Buy ${c} in App`);
+    });
+  });
+
+  it(`should return string "Buy TUSD in App"`, () => {
```
Comment (fjovakaric) [inline app/utils/crypto-util/crypto-util.test.js:148]: change test names for coins that are not in app.

File: app/utils/crypto-util/crypto-util.test.js [hunk 11]
```diff
@@ -0,0 +1,295 @@
...
+  XAUT: "xaut",
+};
+
+describe("isERC20()", () => {
+  Object.values(ERC_20_COINS).forEach(c => {
+    it(`should return true if ${c} is ERC20 coin`, () => {
+      expect(cryptoUtil.isERC20(c)).toBeTruthy();
+    });
+  });
+  const nonErc20Coins = _.differenceWith(
```
[the remark this answers is not in the document]
Comment (fjovakaric) [reply app/utils/crypto-util/crypto-util.test.js:96]: move out of describe

**Review by fjovakaric — APPROVED** [review]
